backend/router/paymentRouter.js

```diff
@@ -3,13 +3,32 @@ const {
   createSubscription,
   getRazorpayApiKey,
   verifySubscription,
+  createPlan,
+  deletePlan,
+  updatePlan,
+  getPlans,
 } = require("../controller/paymentController");
 const jwtAuth = require("../middleware/jwtAuth.js");
+const authAdmin = require("../middleware/authAdmin");
+
 const paymentRouter = express.Router();
 
 paymentRouter.route("/subscribe").post(jwtAuth, createSubscription);
-paymentRouter.route("/razorpaykey").get(jwtAuth, getRazorpayApiKey);
 
-paymentRouter.route("/verify-subscription").post(jwtAuth, verifySubscription);
+paymentRouter.route("/rasorpaykey").get(jwtAuth, getRazorpayApiKey);
+
+paymentRouter
+  .route("/verify-subscription")
+  .post(jwtAuth, authAdmin, verifySubscription);
+
+paymentRouter
+  .route("/plan")
+  .post(jwtAuth, authAdmin, createPlan)
```

Comment:
The auth middleware

Reply:
Check all the auth middlewares in the routes

Reply:
done